### PR TITLE
🔨 Handle the reports workspace selection

### LINKED ISSUE #593: Add the `.selectWorkspaceButtonTapped` action for reports
## What needs to be done?
- [x] Add a "Select workspace" route to the reports feature, which should be display the select workspace pop-up
- [x] Create a `.selectWorkspaceButtonTapped` action. This action should navigate to the newly defined route.
- [x] Add appropriate tests
- [ ] Update the [docs](https://www.notion.so/toggl/Reports-54cac91332aa481db6b1de0d3b90aeb9)

### Dependencies
Depends on #554

### LINKED ISSUE #594: Add the `.workspaceSelected` action for reports
## What needs to be done?
- [x] Create a `.workspaceSelected` action. This action should wrap the newly selected workspace's ID, save that ID to the state, construct a `DateRangeSelection` with `SelectionSource` set to `.initial`, and then emit the `LoadReportsEffect`
- [x] Add appropriate tests
- [ ] Update the [docs](https://www.notion.so/toggl/Reports-54cac91332aa481db6b1de0d3b90aeb9)

### Dependencies
Depends on #563

### LINKED ISSUE #595: Add the select workspace button in reports
## What needs to be done?
- [x] Create a `hasMultipleWorkspaces` selector. It should fetch the workspaces, count them, and return `true` if the user has more than one.
- [x] Create a `currentWorkspaceName` selector. It should fetch the current workspace and return it's name as a String.
- [x] Add the button (designs are pending)
    - [x] Make the button emit the `.selectWorkspaceButtonTapped` action
    - [x] Bind the button's title to the `currentWorkspaceName`
    - [x] Bind the button's visibility to `hasMultipleWorkspaces`
- [x] Add appropriate tests

### Dependencies
Depends on #593

### LINKED ISSUE #596: Add the select workspace dialog for reports
## What needs to be done?
- [x] Create a view that lists all the WSs available to the user, and hightlight the current one
- [x] The view should have a "Close" and "Done" buttons (designs pending)
    - [x] The close button should close the view and trigger no actions
    - [x] The done button should emit a `.workspaceSelected` action with the selected WS's ID and then close the view

### Dependencies
Depends on #594

### PR DESCRIPTION
**Pull request by daividssilverio**
### Summary
~This is WIP to get some opinions.~
~I intend to look into making the single choice composable from the settings view part of the common module and just use it here. What do you think?~ That was done. 

This PR adds the workspace selection logic for the reports page.

### Technical considerations
I've made the workspace picking logic without using our routing system because I think it's easier to go that way.
What do you think?

### Why is it done like this and not some other way?
Because adding a route and a fragment to pick an item doesn't feel good.

### Relationships
Closes #593
Closes #594
Closes #595
Closes #596 

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
You can test it already. The workspace picking logic is working. ~but the data isn't being refreshed for now.~

### Review template
~I'll re-arrange the changes into readable commits when I'm done with it.~
~For now, please just take a look at the changes.~
Review by commit.

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑 it after the draft becomes a PR.

### CODE REVIEW

File: reports/src/main/java/com/toggl/reports/domain/ReportsState.kt [hunk 1]
```diff
@@ -23,13 +23,22 @@ data class ReportsState(
...
             Loadable.Uninitialized,
             null,
-            null
+            null,
+            SelectionViewModel.NotSelecting
         )
     }
 }
+
+sealed class SelectionViewModel<out Value> {
```
Comment (daividssilverio) [inline reports/src/main/java/com/toggl/reports/domain/ReportsState.kt:39]: overkill?

Comment (heytherewill) [reply reports/src/main/java/com/toggl/reports/domain/ReportsState.kt:39]: Nope, it's semantic

File: reports/src/main/java/com/toggl/reports/domain/ReportsWorkspaceSelectionViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.toggl.reports.domain
+
+data class ReportsWorkspaceSelectionViewModel(
+    val currentWorkspaceName: String,
+    val hasMultipleWorkspaces: Boolean
+) {
+    companion object {
+        val Hidden = ReportsWorkspaceSelectionViewModel("", false)
```
Comment (heytherewill) [inline reports/src/main/java/com/toggl/reports/domain/ReportsWorkspaceSelectionViewModel.kt:8]: Why do you hate lowercase static properties? 😄

**Review by heytherewill — COMMENTED** [review]
I like the idea and I definitely think we should extract that single selection picker to a common area so we can have a more unified experience :)

File: reports/src/main/java/com/toggl/reports/domain/ReportsWorkspaceSelectionViewModel.kt (hunk 2 above)
Comment (daividssilverio) [reply reports/src/main/java/com/toggl/reports/domain/ReportsWorkspaceSelectionViewModel.kt:8]: I don't know, they look nicer like this.

File: common-feature/src/main/java/com/toggl/common/feature/compose/common/SingleChoiceListWithHeader.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    Column(modifier = modifier.fillMaxWidth()) {
+        Text(
+            text = header,
+            style = MaterialTheme.typography.h6,
+            color = MaterialTheme.colors.onBackground,
+            modifier = if (description.isEmpty()) Modifier.preferredHeight(grid_8) else Modifier.preferredHeight(
+                grid_5
+            )
+        )
+        if (description != "") {
```
Comment (heytherewill) [inline common-feature/src/main/java/com/toggl/common/feature/compose/common/SingleChoiceListWithHeader.kt:43]:
```suggestion
        if (description.isNotBlank()) {
```

File: reports/src/main/java/com/toggl/reports/ui/composables/ReportsPage.kt [hunk 4]
```diff
@@ -53,26 +61,59 @@ fun ReportsContent(
...
+}
+
+@Composable
+private fun SelectedWorkspaceFAB(
+    modifier: Modifier,
+    label: String,
+    dispatcher: (ReportsAction) -> Unit
+) {
+    ExtendedFloatingActionButton(
+        modifier = modifier.padding(16.dp),
```
Comment (heytherewill) [inline reports/src/main/java/com/toggl/reports/ui/composables/ReportsPage.kt:111]:
```suggestion
        modifier = modifier.padding(grid_2),
```

**Review by heytherewill — APPROVED** [review]

**Pull request merged**